### PR TITLE
allow_reset as constructor argument

### LINKED ISSUE #608: feature request: `allow_reset` as a constructor argument for `ChromaDbConfig`
### 🚀 The feature

Currently we throw 
```
ValueError: For safety reasons, resetting is disabled.Please enable it by including `chromadb_settings={"allow_reset": True}` in your ChromaDbConfig`.
```

It's debatable whether we need this safety mechanism at all.

But either way, allow_reset should be directly configurable under ChromaDbConfig, without the need to pass it as a dict in the "advanced settings".

So the task is to make this accessible from the ChromaDbConfig constructor.

This 

```python
app = App(chromadb_config=ChromaDbConfig(chroma_settings={"allow_reset": True}))
```

should turn into this

```python
app = App(chromadb_config=ChromaDbConfig(allow_reset=True))
```

### Motivation, pitch

allow_reset should be directly configurable under ChromaDbConfig, without the need to pass it as a dict in the "advanced settings".

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Description

Set `allow_reset` as constructor argument in ChromadbConfig.

Fixes #608 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: embedchain/config/vectordbs/ChromaDbConfig.py [hunk 1]
```diff
@@ -27,9 +28,10 @@ def __init__(
         :type port: Optional[str], optional
         :param chroma_settings: Chroma settings dict, defaults to None
         :type chroma_settings: Optional[dict], optional
+        :param allow_reset: Resets the database. defaults to False
```
Comment (cachho) [inline embedchain/config/vectordbs/ChromaDbConfig.py:31]: I'd prefer if you changed the order (I know that could break stuff), but `chroma_settings` is a catch-all solution that should come last.

**Comment by cachho** [comment]
Other than that it looks good. thanks

File: embedchain/vectordb/chroma.py [hunk 2]
```diff
@@ -209,7 +210,7 @@ def reset(self):
         except ValueError:
             raise ValueError(
                 "For safety reasons, resetting is disabled."
```
Comment (cachho) [inline embedchain/vectordb/chroma.py:212]: please include a space after the period.

File: embedchain/config/vectordbs/ChromaDbConfig.py [hunk 3]
```diff
@@ -13,6 +13,7 @@ def __init__(
         host: Optional[str] = None,
         port: Optional[str] = None,
         chroma_settings: Optional[dict] = None,
```
Comment (cachho) [inline embedchain/config/vectordbs/ChromaDbConfig.py:15]: I mean change the order here too

Comment (Dev-Khant) [reply embedchain/config/vectordbs/ChromaDbConfig.py:15]: Ohh Alright

File: embedchain/vectordb/chroma.py (hunk 2 above)
Comment (Dev-Khant) [reply embedchain/vectordb/chroma.py:212]: Where I didn't get it?

**Review by cachho — APPROVED** [review]

**Pull request merged**